### PR TITLE
feat(image): add registry options

### PR DESCRIPTION
**Pull request by knqyf263**
## Description
Refactor registry options to get aligned with other flags and exposes the new flags as below.

```
Registry Flags
      --password strings         password. Comma-separated passwords allowed. TRIVY_PASSWORD should be used for security reasons.
      --registry-token strings   registry token
      --username strings         username. Comma-separated usernames allowed.
```

## Issues
- Close https://github.com/aquasecurity/trivy/issues/2659

## Related PRs
- [x] https://github.com/aquasecurity/trivy/pull/3631

## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [x] I've added tests that prove my fix is effective or that my feature works.
- [x] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [x] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

File: docs/docs/references/customization/config-file.md [hunk 1]
```diff
@@ -163,6 +163,23 @@ db:
   java-repository: ghcr.io/aquasecurity/trivy-java-db
 ```
 
+## Registry Options
```
Comment (DmitriyLewen) [inline docs/docs/references/customization/config-file.md:166]: 1) I think we can add information about these flags to [Docker Private Registries](https://aquasecurity.github.io/trivy/v0.38/docs/advanced/private-registries/)
2) we can also add info about support of multiple credentials.

File: pkg/commands/app.go [hunk 2]
```diff
@@ -304,6 +305,7 @@ func NewFilesystemCommand(globalFlags *flag.GlobalFlagGroup) *cobra.Command {
 		MisconfFlagGroup:       flag.NewMisconfFlagGroup(),
 		ModuleFlagGroup:        flag.NewModuleFlagGroup(),
 		RemoteFlagGroup:        flag.NewClientFlags(), // for client/server mode
+		RegistryFlagGroup:      flag.NewRegistryFlagGroup(),
```
Comment (DmitriyLewen) [inline pkg/commands/app.go:308]: It looks like we are not using credentials in `fs` mode, or am I missing something?

File: pkg/commands/artifact/run.go [hunk 3]
```diff
@@ -648,6 +662,7 @@ func initScannerConfig(opts flag.Options, cacheClient cache.Cache) (ScannerConfi
 				Full: opts.LicenseFull,
 			},
 		},
+		RemoteOption: remoteOption,
```
Comment (DmitriyLewen) [inline pkg/commands/artifact/run.go:665]: It looks we duplicate `remoteOption` in [ArtifactOption](https://github.com/aquasecurity/trivy/pull/3906/files#diff-3b6c25b1435d1150f7492a6c5b47429c0f0bf1f630340582163cece9823168b7R650) and here.
Perhaps we can use `remoteOption` from `ArtifactOption`?

File: pkg/commands/app.go (hunk 2 above)
Comment (knqyf263) [reply pkg/commands/app.go:308]: The credential will be used for downloading DB. I'll raise another PR.

File: pkg/commands/artifact/run.go (hunk 3 above)
Comment (knqyf263) [reply pkg/commands/artifact/run.go]: Great feedback! Fixed in https://github.com/aquasecurity/trivy/pull/3906/commits/554e01c19f380c2374288cdfca138a80b6546ede

File: docs/docs/references/customization/config-file.md (hunk 1 above)
Comment (knqyf263) [reply docs/docs/references/customization/config-file.md:166]: Updated https://github.com/aquasecurity/trivy/pull/3906/commits/3cdc7c2544e7f3d55da1f0d7b13d8716dcd5edb9

**Review by DmitriyLewen — APPROVED** [review]

File: docs/docs/advanced/private-registries/docker-hub.md [hunk 4]
```diff
@@ -1,7 +1,2 @@
-Docker Hub needs `TRIVY_USERNAME` and `TRIVY_PASSWORD`.
-You don't need to set ENV vars when download from public repository.
-
-```bash
-export TRIVY_USERNAME={DOCKERHUB_USERNAME}
```
Comment (AnaisUrlichs) [inline docs/docs/advanced/private-registries/docker-hub.md:5]: Is this specific to DockerHub? Why are we calling it TRIVY_USERNAME and PASSWORD and not DOCKER_USERNAME and PASSWORD?

Comment (AnaisUrlichs) [reply docs/docs/advanced/private-registries/docker-hub.md:5]: Or instead REGISTRY_USERNAME

Comment (knqyf263) [reply docs/docs/advanced/private-registries/docker-hub.md:5]: That's why I removed it from Docker Hub and moved it under private images 😄

Comment (knqyf263) [reply docs/docs/advanced/private-registries/docker-hub.md:5]: No, it is not docker specific. I renamed them.
https://github.com/aquasecurity/trivy/pull/3906/files#diff-a812993c8d91c3eacbf5538b3f3cffaa8a1fe76c48f493caf7882484113d648fR8

**Pull request merged**